test/project_types/theme/commands/pull_test.rb

```diff
@@ -39,6 +40,37 @@ def test_can_pull_theme
         end
       end
 
+      def test_can_specify_env
```

Comment:
Not really a problem with this test, but is there a way to write data providers for minitest? That would really reduce the duplication in our tests.

Reply:
What's a data provider? I've not heard that term it the Ruby world.

Reply:
It's a set of data you use when running a data-driven test. In my experience from the PHP world and PHPUnit specifically, it's a method that returns an array of arrays, like so (roughly translated to ruby):

```
def data_provider_method
  [
    [set1_arg1, set1_arg2],
    [set2_arg1, set2_arg2],
  ]
end

def test_that_uses_provider(arg1, arg2)
  # use arg1 and arg2 here
end
```
It's basically a way of abstracting the specificities of the test, which is particularly helpful when we're testing the exact same scenario under different circumstances (like say, the env flag in this case).

Reply:
I see, there's nothing built-in for that, it would typically be handled with just a private method in the test. Or for something more complex you could have a separate class using the [Builder pattern](http://www.natpryce.com/articles/000714.html).

Reply:
I've also seen suggestions online to dynamically generate `test_` methods, which kind of amounts to the same as using a private method (though the former can fail individually), but that seems slightly over-complicated for a somewhat simple issue.